pdns/ws-auth.cc

```diff
@@ -518,7 +518,7 @@ static void gatherRecords(const Json container, const DNSName& qname, const QTyp
       // verify that there's a zone for the PTR
       SOAData sd;
       if (!B.getAuth(ptr.qname, QType(QType::PTR), &sd, false))
-        throw ApiException("Could not find domain for PTR '"+ptr.qname.toString()+"' requested for '"+ptr.content+"'");
+        throw ApiException(ApiException::ErrZoneNotExists, "Could not find domain for PTR '"+ptr.qname.toString()+"' requested for '"+ptr.content+"'");
```

Comment:
404 Not found is exclusive to HTTP world. Even though this is an HTTP interface, I don't believe error messages should be tied to do. IMO, it is better to have more generic error codes that do have a relation to the application itself rather than to HTTP protocol (moreover Not Found is already in http status message, so don't feel it is really needed here)